### ISSUE TITLE
Multiply with MorehCumSum PCC error

### ISSUE THREAD

**Issue by jserbedzijaTT** [tenstorrent/tt-metal#26573]
### Describe the bug

I get a PCC error when running the following test:
```
# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
# SPDX-License-Identifier: Apache-2.0

import pytest
import torch
import ttnn

from tests.ttnn.utils_for_testing import assert_with_pcc

@pytest.mark.parametrize("device_params", [{"l1_small_size": 32768}], indirect=True)
def test_ir_reshape_cumsum_multiply(device):
    """
    Test that models the following IR sequence:
    1. Reshape from [1, 32] to [1, 32, 1, 1]
    2. Apply moreh_cumsum on dim=1
    3. Reshape back to [1, 32]
    4. Multiply with original input
    """
    
    # Create input tensor [1, 32] with int32 values
    torch.manual_seed(0)
    torch_input = torch.randint(-100000, 100000, (1, 32), dtype=torch.int32)
    
    # Convert to ttnn tensor with int32 dtype
    tt_input = ttnn.from_torch(
        torch_input,
        dtype=ttnn.int32,
        layout=ttnn.TILE_LAYOUT,
        device=device,
        memory_config=ttnn.DRAM_MEMORY_CONFIG
    )
    
    # Operation 1: Reshape to [1, 32, 1, 1]
    tt_reshaped_1 = ttnn.reshape(tt_input, [1, 32, 1, 1])
    
    # Operation 2: Apply moreh_cumsum on dim=1
    tt_cumsum = ttnn.operations.moreh.cumsum(tt_reshaped_1, dim=1)
    
    # Deallocate intermediate tensor (matching IR)
    tt_reshaped_1.deallocate()
    
    # Operation 3: Reshape back to [1, 32]
    tt_reshaped_2 = ttnn.reshape(tt_cumsum, [1, 32])
    
    # Deallocate intermediate tensor (matching IR)
    tt_cumsum.deallocate()
    
    # Operation 4: Multiply with original input
    tt_output = ttnn.multiply(tt_reshaped_2, tt_input, dtype=ttnn.int32)
    
    # Deallocate intermediate tensor (matching IR)
    tt_reshaped_2.deallocate()
    
    # Convert output back to torch for validation
    output = ttnn.to_torch(tt_output)
    
    # Compute reference in PyTorch
    torch_reshaped_1 = torch_input.reshape(1, 32, 1, 1)
    torch_cumsum = torch.cumsum(torch_reshaped_1, dim=1)
    torch_reshaped_2 = torch_cumsum.reshape(1, 32)
    torch_reference = torch_reshaped_2 * torch_input
    
    assert_with_pcc(torch_reference, output, 0.999)
    
    # Clean up
    tt_output.deallocate()
```

**Comment by jserbedzijaTT** [comment]
@cmaryanTT Could someone take a look?

**Comment by cmaryanTT** [comment]
@jserbedzijaTT - can you please do some due diligence on this first and analyze where the PCC loss occurs. There are three OPs that I see here (cumsum, reshape, multiply - all int32)

**Comment by jserbedzijaTT** [comment]
The PCC error occurs after the multiply operation.

**Comment by fahmedchanTT** [comment]
@bbradelTT Would you help triage this ?

**Comment by cmaryanTT** [comment]
@KalaivaniMCW - can you please find someone to test out these steps and confirm what the input shape into multiply is, and why there is a loss of accuracy there. This is all int32, and with the input at the top I would expect a perfect answer (equality comparison).

@aczajkowskiTT - does cumsum support int32? Should they be using the Moreh variant or our variant? (and for that matter, should we remove the Moreh variant, if our variant is a superset)

**Comment by jbbieniekTT** [comment]
@cmaryanTT moreh cumsum has been recently adapted to share common codebase with the regular cumsum and it uses the add tile int32 llk

**Comment by KalaivaniMCW** [comment]
will take a look at this

**Comment by fahmedchanTT** [comment]
@KalaivaniMCW Any update here ?

**Comment by mouliraj-mcw** [comment]
Hi @jserbedzijaTT,

The same test passes with torch.equal but fails with assert_with_pcc.
The issue appears to be with assert_with_pcc, since Torch outputs int64 while ours is int32. The input starts as int32, but Torch converts it to int64 in the torch.cumsum operation.

PS: No issues observed with mul.

cc: @cmaryanTT @fahmedchanTT

**Comment by KalaivaniMCW** [comment]
@aczajkowskiTT  Shall we close this ?

**Comment by aczajkowskiTT** [comment]
@KalaivaniMCW The cumsum versions were merged, and should be working correctly.  Based on the above conversation it appears to be an issue with PyTorch assuming int64.  I think we can close this.

**Comment by skhatriTT** [comment]
@jserbedzijaTT can you please confirm? Thank you

C.C: @sdjordjevicTT

**Comment by fahmedchanTT** [comment]
@aczajkowskiTT Would you link the closing PR ?

Thanks

**Comment by cmaryanTT** [comment]
Assigning to @jserbedzijaTT to validate the findings and close the issue.
